pkg/statussyncer/status/controller/placementbinding_controller.go

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) 2020 Red Hat, Inc.
+// Copyright Contributors to the Open Cluster Management project
+
+package controller
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/apimachinery/pkg/types"
+	policyv1 "open-cluster-management.io/governance-policy-propagator/api/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type placementBindingController struct {
+	client   client.Client
+	log      logr.Logger
+	areEqual func(client.Object, client.Object) bool
+}
+
+func (c *placementBindingController) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.Result, error) {
+	reqLogger := c.log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+
+	instance := &policyv1.PlacementBinding{}
+	err := c.client.Get(ctx, request.NamespacedName, instance)
+	if err != nil {
+		reqLogger.Error(err, "Reconciliation failed")
+		return ctrl.Result{}, err
+	}
+
+	for _, subject := range instance.Subjects {
+		if subject.APIGroup == policyv1.SchemeGroupVersion.Group &&
+			subject.Kind == policyv1.Kind {
+			err := c.updatePolicyStatus(ctx, subject.Name, request.Namespace,
+				instance.PlacementRef.Name, instance.Name)
+			if err != nil {
+				return ctrl.Result{}, err
```

Comment:
in case you want to control the requeue period you can use the same as in leaf hub:
https://github.com/stolostron/leaf-hub-status-sync/blob/main/pkg/controller/generic/generic_status_sync_controller.go#L100